### ISSUE TITLE
When Tabbing through map, focus outline flashes but does not stay lit up

### ISSUE THREAD

**Issue by akrawitz** [Leaflet/Leaflet#6986]
Try to use Tab to advance through focusable elements in a Leaflet map. For example, try with the example map on the Leaflet homepage (https://leafletjs.com/). Elements including the "x" to close the popup, the zoom in ("+") and zoom out ("-") controls, and the bottom-right corner links for "Leaflet" and "OpenStreetMap", will briefly flash the focus ring, but it will then disappear. The element is actually still focused, there is no visible indication. Among other things, this is a nightmare for accessibility.

This seems to be occurring because the `Map` `_handleDOMEvent` function calls `DomUtil.preventOutline` for all `mousedown`, `keypress`, `keyup`, and `keydown` events. `DomUtil.preventOutline` is injecting `outline: none` into the `element.style`. With the caveat that I am not familiar with this code base, I don't understand this approach. If the desire is to remove the focus ring outline, why wouldn't this be done with CSS on `:focus`? This would prevent the brief focus ring flash, and it would allow for it to be overridden in CSS (without resorting to `!important`).

**Comment by Malvoz** [comment]
FWIW I believe this is a WCAG 2.1 [Success Critera 2.4.7 Focus Visible](https://www.w3.org/TR/WCAG21/#focus-visible) failure.

I'm not sure if the behavior is intended, because developers have historically removed outlines. Perhaps [`:focus-visible`](https://github.com/WICG/focus-visible) can motivate change.

**Comment by matthoylu** [comment]
Is this issue still being looked at? I am also trying to meet accessibility standards while using Leaflet but this has become a major issue.

**Comment by Malvoz** [comment]
If you (@matthoylu, and others) wish to restore the focus outline in your own stylesheet as a quickfix, here's a snippet that should cover most browsers:


```css
.leaflet-container:focus,
.leaflet-container :focus {
  outline: -webkit-focus-ring-color auto thin !important; /* Fallback in some browsers where `revert` is not supported. */
  /* Longhand properties:
  outline-color: -webkit-focus-ring-color !important;
  outline-style: auto !important;
  outline-width: thin !important;
  */
  outline: revert !important;
}
```

**Comment by jafin** [comment]
In the src, if you swap replace the handleDOMEvent function to on act only on mousedown this also can work around the issue.

**EXISTING:**

```js
  	_handleDOMEvent: function (e) {
  		if (!this._loaded || skipped(e)) { return; }

  		var type = e.type;

  		if (type === 'mousedown' || type === 'keypress' || type === 'keyup' || type === 'keydown') {
			  // prevents outline when clicking on keyboard-focusable element
  			preventOutline(e.target || e.srcElement);
  		}

  		this._fireDOMEvent(e, type);
  	},
```

**AFTER CHANGE**

```js
  	_handleDOMEvent: function (e) {
  		if (!this._loaded || skipped(e)) { return; }

  		var type = e.type;

  		if (type === 'mousedown' {
			  // prevents outline when clicking on keyboard-focusable element
  			preventOutline(e.target || e.srcElement);
  		}

  		this._fireDOMEvent(e, type);
  	},
```

**Comment by IvanSanchez** [comment]
Closev via #7259
